Makefile

```diff
@@ -12,4 +12,31 @@ coverage:
 .PHONY: clean
 clean:
 	rm coverage.cov
-	
\ No newline at end of file
+
+
+.PHONY: install-golangci-lint
+install-golangci-lint:
+	@echo "Installing golangci-lint..."
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/HEAD/install.sh | sh -s -- -b $$(go env GOPATH)/bin v2.4.0
+	@echo "golangci-lint installed successfully!"
+	@golangci-lint --version
+
+.PHONY: check-golangci-lint
+check-golangci-lint:
+	@which golangci-lint > /dev/null || (echo "golangci-lint not found. Run 'make install-golangci-lint' to install it." && exit 1)
+
+.PHONY: go-lint
+go-lint:
+	@which golangci-lint > /dev/null || (echo "golangci-lint not found. Installing..." && $(MAKE) install-golangci-lint)
+	golangci-lint run --config .golangci.yml
+
+.PHONY: go-lint-fix
+go-lint-fix:
+	@which golangci-lint > /dev/null || (echo "golangci-lint not found. Installing..." && $(MAKE) install-golangci-lint)
+	golangci-lint run --config .golangci.yml --fix
+
+.PHONY: format
+format: go-lint
+
+.PHONY: format-fix
+format-fix: go-lint-fix
```

Comment:
I think these were for backward compatibility in asconfig repo. 
Do we need these here also as well?